src/rpc/blockchain.cpp

```diff
@@ -2219,7 +2223,16 @@ static RPCHelpMan savemempool()
         throw JSONRPCError(RPC_MISC_ERROR, "Unable to dump mempool to disk");
     }
 
-    return NullUniValue;
+    auto node_context = util::AnyPtr<NodeContext>(request.context);
+
+    if (!node_context) {
+        return NullUniValue;
```

Comment:
How could this happen? If a mempool exists, a node context must have existed as well. See `EnsureAnyMemPool` above.

This is dead and confusing code.

Reply:
Checked that `EnsureAnyMemPool` calls `EnsureAnyNodeContext(context)`.
Changed the code accordingly.